agent/bench-scripts/test-bin/rpm

```diff
@@ -16,9 +16,9 @@ fi
 # the RPM) and emitting the RPM version that was found (whatever was given
 # as arguments, if not one of the specific RPMs to mimic).
 if [[ "${3}" == "fio" ]]; then
-    echo "fio-3.21"
+    echo "3.21"
 elif [[ "${3}" == "uperf" ]]; then
-    echo "uperf-1.0.7"
+    echo "1.0.7"
 else
     echo "${3}"
 fi
```

Comment:
This doesn't look right:  in two of the cases you're comparing `${3}` to a tool name/package and returning a version number; in the third case (line 23), I presume that you're going to return the tool/package rather than just a version number...not so?

Reply:
Should it instead by erroring-out, calling for a developer to add another special case?

Reply:
Yes, this does look a bit off: the real rpm would produce an error message for an unknown package and return status 1. I guess it depends on how closely the mocked version should resemble the original in its behavior. The mocked one basically pretends that the package always exists (by always returning status 0) and prints out some conventional "version" (which in some cases, e.g. dpdk-tools, is just the name of the package): the only thing that matters in that case is that the gold file has the same thing in it.

Reply:
> the only thing that matters in that case is that the gold file has the same thing in it.

:-D

> the real rpm would produce an error message for an unknown package and return status 1.

The problem is, this mock has no idea of whether the unrecognized package exists or not, so it doesn't know whether to return an error or a mock'd version number.

I'm OK with the mock always pretending that an unrecognized package exists and returning success, but I think it needs to print out something intelligent (or, at least, intelligible), and I'm concerned that this isn't "it".

Reply:
Resolved by making the mock'd `rpm` behavior as close to the real `rpm` as possible.